CMakeLists.txt

```diff
@@ -2,7 +2,7 @@
 # contributors (https://github.com/scipp)
 cmake_minimum_required(VERSION 3.13)
 project(scipp)
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
```

Comment:
is this actually needed for scipp to compile?
i thought it was only the version of the windows compiler that needed to be increased, not the minimum version in the `CMake` file.
Seems wrong to require C++20 as a minimum

Reply:
This is not required on any system but windows. I felt it was better to be compiling all the different systems in the same way if possible but could easily continue to compile against C++17 on the other systems if that is better. Another option would be to set the `CMAKE_CXX_STANDARD_REQUIRED` flag below this to false which would allow compilers to drop back a standard if they don't support the specified one.

In answer to your question above switching to compile against C++20 would mean that the minimum required version of gcc would switch from 7.5 to 8 currently.

Reply:
I think we should not require C++20 if we are not using any features from the 20 standard, and it's only because of a bug in a windows compiler.
Maybe an alternative is to write a note in the developer documentation. If you are developing under windows and you have version x.xx of the compiler, then you need to switch to C++20.
Can you get the windows builder to use the latest compiler without requiring C++20 here?

Reply:
Yes can require C++20 for just windows. Think it should even be possible to detect compiler version and switch which standard you are building against based on that. I'll update to-do that and add some additional explanatory notes for windows to the getting started section.

Reply:
> Yes can require C++20 for just windows. Think it should even be possible to detect compiler version and switch which standard you are building against based on that. I'll update to-do that and add some additional explanatory notes for windows to the getting started section.

I think it would be enough just to keep the cmake flags as they are and set accordingly in the azure configuration files. You might add a documentation to the developer section about using changing the standard according to the VS version.

Reply:
I prefer updating the cmake configuration here for two reasons. The first is that it means that a local windows build will match the CI build as closely as possible and the second is that if we don't update the cmake local windows builds will continue to not work without additional developer steps not captured in the cmake configuration and whilst it is not a common occurrence I think making getting a local build up and running as easy as possible is a nice thing to have.

Reply:
But these are minor points so happy with just updating the azure config if that is preferable.

Reply:
I think it's easy enough if it is clearly stated in the documentation.
Here, you are potentially affecting more users because many people are not on windows.
Finally, the HPC resources at ESS will be running CentOS, and they are usually using relatively old versions of packages/compilers. So i don't know if a C++20 is available.

Reply:
I was making the case here for updating the cmake config for just windows, sorry should have made that clearer. You have persuaded me that updating the other systems as well is not a good idea. I think the benefits are minor however so as I said happy to just update the azure config.